planner/core/find_best_task.go

```diff
@@ -277,6 +277,13 @@ func compareBool(l, r bool) int {
 // If `x` is not worse than `y` at all factors,
 // and there exists one factor that `x` is better than `y`, then `x` is better than `y`.
 func compareCandidates(lhs, rhs *candidatePath) int {
+	// Do NOT prune any other access path if the path is table scan the tiflash.
+	if rhs.path.storeType == kv.TiFlash {
```

Comment:
Can we not do it here? It makes this function looks pretty werid. Can we just append it in `skylinePruning`? I think this is not the only case that need not to be pruned.

Reply:
OK, I'll put it in the `skylinePruning`. BTW, a `DataSource` only generates at most one TiFlash access path.